flow/Hostname.actor.cpp

```diff
@@ -19,9 +19,24 @@
  */
 
 #include "flow/Hostname.h"
+
+#include <regex>
+
 #include "flow/UnitTest.h"
+
 #include "flow/actorcompiler.h" // has to be last include
 
+namespace {
+
+const static std::regex validation("^([\\w\\-]+\\.?)+:([\\d]+){1,}(:tls)?$");
```

Comment:
I think we should consider using `boost::regex` instead. The boost implementation was much faster the last time I looked.

Reply:
No idea, but there are so many regex implementations, e.g. boost::regex, boost::expressive, pcre, re2, etc. I had a benchmark of regex several years ago, https://github.com/xis19/regexp_benchmark, and it seems re2 is having the best performance.

Reply:
Maybe consider switching to some other libraries, but it would be another story.

Reply:
sure, other implementations might be faster, but we already have boost. Switching to boost is as simple as `s/std/boost` -- switching to re2 would be much more involved (as it pulls in another dependency)